apptools/preferences/tests/test_scoped_preferences.py

```diff
@@ -42,14 +42,18 @@ class ScopedPreferencesTestCase(PreferencesTestCase):
     def setUp(self):
         """ Prepares the test fixture before each test method is called. """
 
-        self.preferences = ScopedPreferences()
+        # A temporary directory that can safely be written to.
+        self.tmpdir = tempfile.mkdtemp()
```

Comment:
_Side comment_: should we consider creating an issue to replace all `tempfile.mkdtemp` and `tempfile.mktemp` with more modern `tempfile.TemporaryFile` and `tempfile.TemporaryDirectory` routines?

Reply:
I'm not seeing a reason to change uses of `tempfile.mkdtemp`; for sure it would be nice to have, but there's nothing wrong with `mkdtemp` and general code modernization isn't really something we have bandwidth for. Uses of `tempfile.mktemp` would definitely need to be fixed, since it's deprecated and insecure, but I'm not seeing any uses in the codebase.